fili-core/src/main/java/com/yahoo/bard/webservice/web/ChainingResultSetMapper.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2020 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+
+package com.yahoo.bard.webservice.web;
+
+import com.yahoo.bard.webservice.data.Result;
+import com.yahoo.bard.webservice.data.ResultSet;
+import com.yahoo.bard.webservice.data.ResultSetSchema;
+
+import com.yahoo.bard.webservice.data.metric.LogicalMetricInfo;
+import com.yahoo.bard.webservice.data.metric.mappers.NoOpResultSetMapper;
+import com.yahoo.bard.webservice.data.metric.mappers.RenamableResultSetMapper;
+import com.yahoo.bard.webservice.data.metric.mappers.ResultSetMapper;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This Mapper provides ability to chain additional ResultSetMappers along with the mappers from dependent metric.
+ *
+ */
+public class ChainingResultSetMapper  extends ResultSetMapper implements RenamableResultSetMapper {
+
+    List<ResultSetMapper> chainedResultSetMappers;
+    public static final NoOpResultSetMapper NO_OP_MAPPER = new NoOpResultSetMapper();
+
+    /**
+     * Constructor.
+     *
+     * @param chainedResultSetMappers  The Combined list of ResultSetMappers from dependent metric and
+     *                                 additional Mappers to be chained.
+     */
+    public ChainingResultSetMapper(List<ResultSetMapper> chainedResultSetMappers) {
+        this.chainedResultSetMappers = chainedResultSetMappers;
+    }
+
+    /**
+     * Utility method to Create and Rename the ResultSetMappers in the chained List.
+     *
+     * @param info  LogicalMetricInfo used to rename the mapper with final metric name.
+     * @param mappers  The List of mappers from dependent metrics and additional mappers to be chained in list.
+     *
+     * @return  The The Chained ResultSetMapper
+     */
+    public static ResultSetMapper createAndRenameResultSetMapperChain(LogicalMetricInfo info,
+                                                                      ResultSetMapper... mappers) {
+        List<ResultSetMapper> chainedResultSetMappers = Arrays.asList(mappers);
+
+        return getRenamedMappers(chainedResultSetMappers, info.getName());
```

Comment:
If some of the mappers are chains, unpack them.
If some of the mappers are NoOps, filter them out.